frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -2444,252 +2489,292 @@ ApplicabilityResult instantiateSignature(Context* context,
                                       where,
                                       needsInstantiation,
                                       /* instantiatedFrom */ sig,
-                                      /* parentFn */ parentFnTyped,
-                                      std::move(formalsInstantiated));
+                                      /* parentFn */ parentSignature,
+                                      std::move(formalsInstantiated),
+                                      sig->outerVariables());
 
   // May need to resolve the body at this point to compute final TFS.
-  if (ensureBodyIsResolved(context, call, result)) {
-    if (!result->untyped()->isCompilerGenerated()) {
-      if (isTfsForInitializer(result)) {
-        auto resolvedFn = resolveInitializer(context, result, poiScope);
-        auto newTfs = resolvedFn->signature();
-        if (newTfs->needsInstantiation()) {
-          context->error(newTfs->id(), "Failure to resolve initializer");
-        } else {
-          result = newTfs;
-        }
-      } else {
-        CHPL_ASSERT(false && "Not handled yet!");
-        std::ignore = resolveFunction(context, result, poiScope);
-      }
-    }
+  if (result->isInitializer() && !result->isCompilerGenerated()) {
+    auto rf = resolveFunction(rc, result, poiScope);
+    result = rf->signature();
```

Comment:
Oh, is it about the type of `this` ? Because initializers are methods, in a way? They certainly aren't regular methods and IMO it's an implementation choice to view them as having a regular `this` at all. Anyway I'd prefer to see something more along the lines of `inferOutFormals` here. We have certain methods (initializers) where occasionally (when they are generic) we need to compute the type of `this` after resolving the body. This is very similar to the `out` case.

Could you add some TODO comments here about getting it working in this case more like `inferOutFormals` ?